### PR TITLE
create a separated debuggable scheduler container

### PR DESCRIPTION
**Pull request by YamasouA**
#### What type of PR is this?

/area simulator
/kind cleanup

#### What this PR does / why we need it:
I try to Separate API endpoints and Scheduler.
This PR can build debuggablescheduler Pod.
(But still not work. Only build)
After this PR I try to delete scheduler functions from Server and change endpoints functions to operate scheduler Pod.
#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/318

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test

Will take a look during GW

File: simulator/pkg/debuggablescheduler/debuggable_scheduler.go [hunk 1]
```diff
@@ -49,7 +49,7 @@ func CreateOptionForOutOfTreePlugin(outOfTreePluginRegistry runtime.Registry, pl
 
 	var versionedcfg *v1.KubeSchedulerConfiguration
 	var err error
-	if configFile == nil {
+	if configFile == nil || *configFile == "" {
```
Comment (sanposhiho) [inline simulator/pkg/debuggablescheduler/debuggable_scheduler.go:52]: Can you make a separate PR for this line change?

Comment (YamasouA) [reply simulator/pkg/debuggablescheduler/debuggable_scheduler.go]: I separate PR.
fix this PR.
[60eff56](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/350/commits/60eff567ca22304a30400c753cc2c9239c70ac5a)
configFIle fix PR.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/351

File: simulator/Makefile [hunk 2]
```diff
@@ -23,7 +23,9 @@ mod-download: ## Downloads the Go module
 .PHONY: build
 build:  
 	go build -o ./bin/simulator ./cmd/simulator/simulator.go
+	go build -o ./bin/scheduler ./cmd/scheduler/scheduler.go
 
 .PHONY: start
 start: build
 	./hack/start_simulator.sh
+	./bin/scheduler
```
Comment (sanposhiho) [inline simulator/Makefile:31]: Ah, I think we can just remove start because now the simulator must be running in the docker. We cannot run it locally anymore

File: simulator/cmd/simulator/Dockerfile
Comment (sanposhiho) [inline simulator/cmd/simulator/Dockerfile:1]: Please also modify -
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/179cfc4cc6abfc8584f0ae59b3dc7a0b1f1f86ea/cloudbuild.yaml#L11

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test
/retitle create a separated debuggable scheduler container

File: simulator/cmd/simulator/Dockerfile
Comment (YamasouA) [reply simulator/cmd/simulator/Dockerfile:1]: Thank you for pointing this out.
I forgot to correct the file. ([44cf8bd](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/350/commits/44cf8bd1cd02a3f50efd39b83dc244543a6c9785))

File: simulator/Makefile (hunk 2 above)
Comment (YamasouA) [reply simulator/Makefile]: I fix [11e2311](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/350/commits/11e2311dfcedf74eb3ad1909d8ec0a76a0bccd5a).

**Review by sanposhiho — COMMENTED** [review]
can you also modify -
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/48bdaf9be828b710539a9b60fc844cdcb7710c6a/simulator/docs/running-simulator.md

File: simulator/docs/running-simulator.md
Comment (sanposhiho) [inline simulator/docs/running-simulator.md:1]: Replace L1-L8 with -

```md
## Running simulator

The simulator requires docker installed in your laptop.
We have [docker-compose.yml](../../docker-compose.yml) to run the simulator easily.
```

File: simulator/cmd/scheduler/scheduler.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+	_ "k8s.io/component-base/metrics/prometheus/version" // for version metric registration
+	"k8s.io/klog"
+
+	"sigs.k8s.io/kube-scheduler-simulator/simulator/pkg/debuggablescheduler"
+)
+
+func main() {
+	command, cancelFn, err := debuggablescheduler.NewSchedulerCommand()
+	if err != nil {
+		klog.Info(fmt.Sprintf("failed to build the scheduler command: %+v", err))
```
Comment (sanposhiho) [inline simulator/cmd/scheduler/scheduler.go:19]:
```suggestion
		klog.Info(fmt.Sprintf("failed to build the debuggablescheduler command: %+v", err))
```

File: simulator/docs/running-simulator.md
Comment (YamasouA) [reply simulator/docs/running-simulator.md:1]: I replaced text
[9f03809](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/350/commits/9f03809f443082c4516337cf40d2b835687ab22e)
